File: trade/src/main/java/com/ez08/trade/ui/BaseActivity.java
package com.ez08.trade.ui;

import android.app.ProgressDialog;
import android.content.Context;
import android.content.res.ColorStateList;
import android.os.Bundle;
import android.os.Looper;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    protected Context context;

    protected void dismissBusyDialog() {
        if (pDialog != null) {
            pDialog.dismiss();
        }
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = this;
    }

    protected ColorStateList setTextColor(int color) {
        return context.getResources().getColorStateList(color);
    }

    ProgressDialog pDialog;

    protected void showBusyDialog() {
        pDialog = new ProgressDialog(this);
        pDialog.setMessage("请稍候...");
        pDialog.setCancelable(true);
        pDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
        if (Thread.currentThread() == Looper.getMainLooper().getThread()) {
            pDialog.show();
        } else {
            runOnUiThread(() -> pDialog.show());
        }
    }
}

File: app/src/main/java/com/god/kotlin/BaseActivity.kt
package com.god.kotlin

import android.app.ProgressDialog
import android.content.Context
import android.os.Bundle
import android.os.Looper
import android.view.MotionEvent
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.ez08.trade.exception.SessionLostException
import com.ez08.trade.net.Client
import com.ez08.trade.net.callback.StateListener

open class BaseActivity : AppCompatActivity() {

    protected lateinit var context: Context
    private lateinit var stateListener: StateListener

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context = this
        stateListener = object :StateListener{
            override fun connect() {
                onConnect()
            }

            override fun exchange() {
                onExchange()
            }

            override fun login() {
                onLogin()
            }

            override fun kickOut() {
                onKickOUt()
            }

            override fun disconnect(e: Exception) {
                onDisConnect(e)
            }
        }

        Client.getInstance().registerListener(stateListener)
    }

    override fun onDestroy() {
        dismissBusyDialog()
        super.onDestroy()
        Client.getInstance().unRegisterListener(stateListener)
    }

    protected open fun onConnect(){

    }

    protected open fun onExchange(){

    }

    protected open fun onLogin(){

    }

    protected open fun onKickOUt(){
        finish()
    }

    protected open fun onDisConnect(e :Exception){
        if(e is SessionLostException){
            finish()
        }
    }

    public fun dismissBusyDialog() {
        if (pDialog != null && pDialog!!.isShowing) {
            pDialog?.dismiss()
        }
    }

    private var pDialog: ProgressDialog? = null

    public fun showBusyDialog() {
        pDialog = ProgressDialog(this)
        pDialog.let {
            it?.setMessage("请稍候...")
            it?.setCancelable(true)
            it?.setProgressStyle(ProgressDialog.STYLE_SPINNER)
            it?.show()
        }
    }

}